src/main/kotlin/com/vanniktech/maven/publish/nexus/CloseAndReleaseRepositoryTask.kt

```diff
@@ -0,0 +1,26 @@
+package com.vanniktech.maven.publish.nexus
+
+import com.vanniktech.maven.publish.MavenPublishPluginExtension
+import com.vanniktech.maven.publish.findMandatoryProperty
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+
+open class CloseAndReleaseRepositoryTask : DefaultTask() {
+  @SuppressWarnings("unused")
+  @TaskAction
+  fun closeAndReleaseRepository() {
+    val mavenPublishPluginExtension = project.extensions.getByType(MavenPublishPluginExtension::class.java)
+    val nexusOptions = mavenPublishPluginExtension.nexusOptions
+
+    val baseUrl = nexusOptions.baseUrl ?: OSSRH_API_BASE_URL
+    val groupId = nexusOptions.groupId ?: project.findMandatoryProperty("GROUP")
+    val repositoryUsername = nexusOptions.repositoryUsername ?: project.findMandatoryProperty("SONATYPE_NEXUS_USERNAME")
+    val repositoryPassword = nexusOptions.repositoryPassword ?: project.findMandatoryProperty("SONATYPE_NEXUS_PASSWORD")
```

Comment:
When publishing we're also checking the environment variables for `SONATYPE_NEXUS_...`. See `DEFAULT_TARGET` in `MavenPublishPluginExtension`.